lib/CoreBot.js

```diff
@@ -638,6 +638,28 @@ function Botkit(configuration) {
         return this;
     };
 
+    botkit.hearsIntent = function(intent_names, confidence, events, cb) {
+        if (!botkit.witbot) {
+            botkit.log('Please set token_wit in controller.spawn()');
+            return;
+        }
+
+        if (typeof(intent_names) == 'string') {
+            intent_names = intent_names.split(/\,/g);
+        }
+
+        // This is based off of https://www.npmjs.com/package/witbot#example
+        botkit.hears('.*', events, function(bot, message) {
+            var wit = botkit.witbot.process(message.text, bot, message);
+
+            for (index in intent_names) {
```

Comment:
You assume `intent_names` is an array, but it can be a object, variable etc. Can you write more defensive?